src/AppBundle/Resources/public/js/app.deck.js

```diff
@@ -545,68 +543,46 @@ deck.display = function display(container, options) {
 deck.get_layout_data = function get_layout_data(options) {
 
 	var data = {
-			image1: '',
-			image2: '',
-			meta: '',
-			upgrades: '',
-			events: '',
-			allies: '',
-			permanent: '',
-			player_side_schemes: '',
-			supports: '',
-			resources: '',
-			cards: '',
-			hero_color_1: '',
-			hero_color_2: '',
-			hero_color_3: ''
+		image1: '',
+		image2: '',
+		meta: '',
+		upgrades: '',
+		events: '',
+		allies: '',
+		permanent: '',
+		player_side_schemes: '',
+		supports: '',
+		resources: '',
+		cards: '',
+		hero_color_1: '',
+		hero_color_2: '',
+		hero_color_3: ''
 	};
 
-	//var investigator = deck.get_investigator();
 	var problem = deck.get_problem();
 	$("input[name=problem]").val(problem);
 
-	var card = app.data.cards.findById(this.get_investigator_code());
-	var size = 30;
-	var req_count = 0;
-	var req_met_count = 0;
+	var hero = app.data.cards.findById(hero_code);
+	var minDeckSize = 40;
 
-	if (card && card.deck_requirements){
-		if (card.deck_requirements.size){
-			size = card.deck_requirements.size;
-		}
-		// must have the required cards
-		if (card.deck_requirements.card){
-			$.each(card.deck_requirements.card, function (key, value){
-				req_count++;
-				var req = app.data.cards.findById(value);
-				if (req && req.indeck){
-					req_met_count++;
-				}
-			});
-			if (req_met_count < req_count){
-				//return "investigator";
-			}
-		}
-	}
-
-	if (card.meta && card.meta.colors) {
-		data['hero_color_1'] = card.meta.colors[0];
-		data['hero_color_2'] = card.meta.colors[1];
-		data['hero_color_3'] = card.meta.colors[2];
+	if (hero.meta?.colors) {
```

Comment:
fyi, the code is so old, the modern js things like the optional chain ? do not compile when deploying to production. (same goes for using const instead of var). I can merge and fix though.

Reply:
Thanks for fixing those after the merge. Didn't realize running it locally would be different than when it compiles for production.